openedx/core/djangoapps/agreements/models.py

```diff
@@ -22,3 +22,18 @@ class IntegritySignature(TimeStampedModel):
     class Meta:
         app_label = 'agreements'
         unique_together = ('user', 'course_key')
+
+
+class LTIPIITool(models.Model):
+    """
+    This model stores the relationship between a course and the LTI tools in the course that share PII.
+    """
+    user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
+    id = models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')
+    course_key = CourseKeyField(max_length=255, db_index=True)
+    lti_tools = models.JSONField(default='{}')
```

Comment:
When you define a default for a `JSONField`, you want to be sure to use a callable. Take a look at the [Django JSONField documentation](https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.JSONField).

> If you give the field a [default](https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.Field.default), ensure it’s a callable such as the [dict](https://docs.python.org/3/library/stdtypes.html#dict) class or a function that returns a fresh object each time. Incorrectly using a mutable object like default={} or default=[] creates a mutable default that is shared between all instances.

Do you think we need the default? This makes supplying a list of LTI tools optional, and I don't think it should be. Let me know what you think!

Reply:
That makes sense! I was thinking about what happens if there aren't any tools.

Reply:
Yes, an empty table sounds good! That way we are only storing what exists.